<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find multi_nav)/maps/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="tb0" default="tb3_0"/>
  <arg name="tb1" default="tb3_1"/>

  <!-- Map Server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="map"/>
  </node>

  <group ns="$(arg tb0)">
    <!-- Launch Turtlebot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="multi_robot_name" value="$(arg tb0)" />
    </include>

    <!-- Launch an AMCL node -->
    <node pkg="amcl" type="amcl" name="amcl">
      <rosparam file="$(find multi_nav)/param/amcl.yaml" command="load" />
      <param name="initial_pose_x"    value="0.0"/>
      <param name="initial_pose_y"    value="0.0"/>
      <param name="initial_pose_a"    value="0.0"/>
      <param name="odom_frame_id"     value="$(arg tb0)/odom"/>
      <param name="base_frame_id"     value="$(arg tb0)/base_footprint"/>
      <param name="global_frame_id"   value="map"/>
      <param name="use_map_topic"     value="true" />
    </node>
  </group>

  <group ns="$(arg tb1)">
    <!-- Launch Turtlebot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="multi_robot_name" value="$(arg tb1)" />
    </include>

    <!-- Launch an AMCL node -->
    <node pkg="amcl" type="amcl" name="amcl">
      <rosparam file="$(find multi_nav)/param/amcl.yaml" command="load" />
      <param name="initial_pose_x"    value="1.0"/>
      <param name="initial_pose_y"    value="0.5"/>
      <param name="initial_pose_a"    value="0.0"/>
      <param name="odom_frame_id"     value="$(arg tb1)/odom"/>
      <param name="base_frame_id"     value="$(arg tb1)/base_footprint"/>
      <param name="global_frame_id"   value="map"/>
      <param name="use_map_topic"     value="true" />
    </node>
  </group>

  <!-- Launch global position node -->
  <node name="global_position2" pkg="multi_nav" type="global_position2" output="screen">
        <param name="tb3_0" value="$(arg tb0)" />
        <param name="tb3_1" value="$(arg tb1)" />
    </node>

  <!-- Launch a move_base node -->
  <include file="$(find multi_nav)/launch/includes/move_base2.launch">
    <arg name="tb0" value="$(arg tb0)" />
    <arg name="tb1" value="$(arg tb1)" />
  </include>

  <!-- Launch rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find multi_nav)/rviz/turtlebot3_navigation2.rviz"/>
  </group>

</launch>